### DISCUSSION TITLE
Make `source-map-support` opt-in or provide a way to opt out

### DISCUSSION THREAD

**Discussion by lforst** [remix-run/remix#9752] [Proposals]
### Background

`source-map-support`, which is used by `@remix-run/serve` does a pretty invasive change by changing the `Error.stack` property to include the original file names and paths. This makes it very hard for observability tools to properly source map production errors with source code context, since these tools need to map the original stack trace to source files and source maps provided by the user.

### Proposal

- Either make `source-map-support` opt-in. (disclaimer: I can already see that this would degrade peoples DX)
- Provide a way to opt out of using `source-map-support` so that error monitoring tools can pick up original stack traces.

**Comment by lforst** [comment]
Opened a PR in case we want to speedrun this: https://github.com/remix-run/remix/pull/9753

**Reply by brophdawg11** [reply]
The way to opt-out is to use the Remix express adapter where you have full control over your `server.js` file - https://remix.run/docs/en/main/other-api/serve

> It's a production-ready, but basic Node.js server built with Express. If you find you want to customize it, use the @remix-run/express adapter instead.
